zerver/lib/actions.py

```diff
@@ -557,7 +562,9 @@ def revoke_preregistration_users(
 
     if used_preregistration_user is not None:
         used_preregistration_user.status = confirmation_settings.STATUS_ACTIVE
```

Comment:
@mateuszmandera @alexmv do we want this PR to create a new `STATUS_USER_CREATED` value that we should start setting here, given that `STATUS_ACTIVE` is a value that we set when a link is accessed, regardless of whether a user is actually created?

I don't think we have a good way to backfill / deal with old rows with `STATUS_ACTIVE`, but at least we can make it super clear for new ones that have a `created_user_profile` attached?

Reply:
This isn't accurate -- we don't set the invitation to `STATUS_ACTIVE` when the link is clicked.  `check_prereg_key` calls `get_object_from_key(..., activate_object=False)` since 4680d504de03cf30e593f3454e489628d134a735; see #12281.